src/main/resources/application.yml

```diff
@@ -0,0 +1,19 @@
+spring:
+  cloud:
+    gateway:
+      routes:
+      - id: catalog_route
+        uri: http://localhost:8081
+        predicates:
+        - Path=/api/**
+        filters:
+        - name: Retry
+          args:
+            retries: 3
+            statuses: BAD_GATEWAY
+            methods: GET,POST
```

Comment:
GET is safe to retry because we know it doesn't have side effects, but POST can cause trouble, so let's retry only GEY by default, and we will decide for other methods later